satellite/audit/reservoir.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"math/rand"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// Reservoir holds a certain number of segments to reflect a random sample
+type Reservoir struct {
+	Paths []storj.Path
```

Comment:
I remember a comment from @mobyvb that we may want to have this type generic, so it may be used by other services in the future.

Do we still want to do this? In such case, instead of `Paths []storj.Path`, we should have `Items []interface{}`. And, perhaps, move the type to a separate package under /pkg/reservoirs.

Reply:
Do not make things unless we have a use-case, because it will end up using more memory and adds more pressure to the GC. I would even go as far as hardcoding the maximum number of paths this reservoir can hold rather than using a slice.

Reply:
Oh, somehow I thought the compiler will optimize the code and the machine code would be more or less the same. If there is a penalty, let's forget about the generalization.

Reply:
Both `interface{}` and a slice create an indirection.